### PR TITLE
Changes around cubic bezier line intersections

### PR DESCRIPTION
**Pull request by kleintom**
(I haven't checked the behavior when a curve and a line segment intersect at an endpoint: I _think_ the old behavior depended on the "left/top closed, right/bottom open" behavior of Rect.contains, and maybe rounding issues; now I think we have a "left closed, right closed" behavior (and still rounding issues?), so we might include an endpoint intersection in some cases where we didn't previously.  Maybe a new issue? or I can address it here if you like.)

### CODE REVIEW

**Comment by kleintom** [comment]
> I haven't checked the behavior when a curve and a line segment intersect at an endpoint

(Ah, `line_intersections_t` excludes `t = 0` and `t = 1`, so that already excludes curve/line segment endpoint/endpoint intersections.  You could still get line segment endpoint intersecting curve interior I guess, but not the other way around.)

File: geom/src/utils.rs [hunk 1]
```diff
@@ -55,13 +55,13 @@ pub fn cubic_polynomial_roots<S: Scalar>(a: S, b: S, c: S, d: S) -> ArrayVec<[S;
 
     if S::abs(a) < S::EPSILON {
         // quadratic equation
-        let delta = b * b - S::FOUR * a * c;
+        let delta = c * c - S::FOUR * b * d;
```
Comment (nical) [inline geom/src/utils.rs]: Oops!

File: geom/src/cubic_bezier.rs [hunk 2]
```diff
@@ -612,16 +612,37 @@ impl<S: Scalar> CubicBezierSegment<S> {
...
+        } else {
+            segment.bounding_range_x()
+        };
+
         for t in intersections {
-            if aabb.contains(&self.sample(t)) {
-                let t2 = (self.sample(t) - segment.from).length() / segment.length();
-                result.push((t,t2));
+            if seg_is_mostly_vertical {
+                let intersection_y = self.y(t);
```
Comment (nical) [inline geom/src/cubic_bezier.rs]: Nit: How about

```rust
let intersection_xy = if seg.is_mostly_vertical { self.y(t) } else { self.x(t) };
if intersection_xy >= seg_long_axis_min && intersection_xy <= seg_long_axis_max {
   let t2 = (self.sample(t) - segment.from).length() / segment.length();
   result.push((t, t2));
}
```

**Comment by nical** [comment]
Looks like this needs to be rebased and then we're good to go.

**Pull request merged**